v2/examples/pointuh/main.go

```diff
@@ -0,0 +1,265 @@
+		return fmt.Errorf("--output-pkg must be specified")
+	}
+	if len(args.outputFile) == 0 {
+		return fmt.Errorf("--output-file must be specified")
+	}
+
+	return nil
+}
+
+// getNameSystems returns the name system used by the generators in this package.
+func getNameSystems() namer.NameSystems {
+	return namer.NameSystems{
+		"raw": namer.NewRawNamer("", nil),
+	}
+}
+
+// getDefaultNameSystem returns the default name system for ordering the types to be
+// processed by the generators in this package.
+func getDefaultNameSystem() string {
+	return "raw"
+}
+
+// getTargets is called after the inputs have been loaded.  It is expected to
+// examine the provided context and return a list of Packages which will be
+// executed further.
+func getTargets(c *generator.Context, args *Args) []generator.Target {
+	boilerplate, err := gengo.GoBoilerplate(args.goHeaderFile, gengo.StdBuildTag, gengo.StdGeneratedBy)
+	if err != nil {
+		klog.Fatalf("failed loading boilerplate: %v", err)
+	}
+
+	var targets []generator.Target
+	for _, input := range c.Inputs {
+		klog.V(2).InfoS("processing", "pkg", input)
+
+		pkg := c.Universe[input]
+
+		targets = append(targets, &generator.SimpleTarget{
+			PkgName: pkg.Name,
+			PkgPath: filepath.Join(args.outputPkg, pkg.Name),
+			PkgDir:  filepath.Join(args.outputDir, filepath.Base(pkg.Path)),
+
+			HeaderComment: boilerplate,
+
+			// FilterFunc returns true if this Package cares about this type.
+			// Each Generator has its own Filter method which will be checked
+			// subsequently.  This will be called for every type in every
+			// loaded package, not just things in our inputs.
+			FilterFunc: func(c *generator.Context, t *types.Type) bool {
+				// Only consider types in our inputs
+				return t.Name.Package == pkg.Path
+			},
+
+			// GeneratorsFunc returns a list of Generators, each of which is
+			// responsible for a single output file (though multiple generators
+			// may write to the same one).
+			GeneratorsFunc: func(c *generator.Context) (generators []generator.Generator) {
+				return []generator.Generator{
+					newPointuhGenerator(args.outputFile, pkg),
+				}
+			},
+		})
+	}
+
+	return targets
+}
+
+// pointuhGenerator produces a file with autogenerated functions.
+type pointuhGenerator struct {
+	generator.GolangGenerator
+	myPackage *types.Package
+}
+
+func newPointuhGenerator(outputFilename string, pkg *types.Package) generator.Generator {
+	return &pointuhGenerator{
+		GolangGenerator: generator.GolangGenerator{
+			OutputFilename: outputFilename,
+		},
+		myPackage: pkg,
+	}
+}
+
+// Namers returns a set of NameSystems which will be merged with the namers
+// provided when executing this package. In case of a name collision, the
+// values produced here will win.
+func (g *pointuhGenerator) Namers(*generator.Context) namer.NameSystems {
+	return namer.NameSystems{
+		// This elides package names when the name is in "this" package.
+		"localraw": namer.NewRawNamer(g.myPackage.Path, nil),
```

Comment:
I've seen this pattern in a few examples. Is this something that could make a sensible default? I feel like the local-aware namer and per-target import trackers are something consumers of this library wouldnt mind being plumbed for them?

Reply:
I do think we could simplify how namers work more - I didn't cover it in this PR because it was already too big.